vendors/nxp/boards/lpc54018iotmodule/CMakeLists.txt

```diff
@@ -0,0 +1,317 @@
+    ${lpc54018_dir}/utilities/virtual_com.c
+)
+
+set(drivers_src
+    ${lpc54018_dir}/drivers/fsl_clock.c
+    ${lpc54018_dir}/drivers/fsl_common.c
+    ${lpc54018_dir}/drivers/fsl_dma.c
+    ${lpc54018_dir}/drivers/fsl_emc.c
+    ${lpc54018_dir}/drivers/fsl_flexcomm.c
+    ${lpc54018_dir}/drivers/fsl_gpio.c
+    ${lpc54018_dir}/drivers/fsl_inputmux.c
+    ${lpc54018_dir}/drivers/fsl_pint.c
+    ${lpc54018_dir}/drivers/fsl_power.c
+    ${lpc54018_dir}/drivers/fsl_reset.c
+    ${lpc54018_dir}/drivers/fsl_sha.c
+    ${lpc54018_dir}/drivers/fsl_spi.c
+    ${lpc54018_dir}/drivers/fsl_spi_dma.c
+    ${lpc54018_dir}/drivers/fsl_spifi.c
+    ${lpc54018_dir}/drivers/fsl_spifi_dma.c
+    ${lpc54018_dir}/drivers/fsl_usart.c
+    ${lpc54018_dir}/drivers/mflash_drv.c
+    ${lpc54018_dir}/drivers/mflash_file.c
+)
+
+set(lpc54018_includes
+    "${lpc54018_dir}"
+    "${lpc54018_dir}/drivers"
+    "${lpc54018_dir}/cmsis_drivers"
+    "${lpc54018_dir}/middleware/wifi_qca"
+    "${lpc54018_dir}/middleware/wifi_qca/common_src/hcd"
+    "${lpc54018_dir}/middleware/wifi_qca/common_src/include"
+    "${lpc54018_dir}/middleware/wifi_qca/common_src/stack_common"
+    "${lpc54018_dir}/middleware/wifi_qca/common_src/wmi"
+    "${lpc54018_dir}/middleware/wifi_qca/custom_src/include"
+    "${lpc54018_dir}/middleware/wifi_qca/custom_src/stack_custom"
+    "${lpc54018_dir}/middleware/wifi_qca/include"
+    "${lpc54018_dir}/middleware/wifi_qca/include/AR6002"
+    "${lpc54018_dir}/middleware/wifi_qca/include/AR6002/hw2.0/hw"
+    "${lpc54018_dir}/middleware/wifi_qca/port"
+    "${lpc54018_dir}/middleware/wifi_qca/port/boards/lpc54018iotmodule/freertos"
+    "${lpc54018_dir}/middleware/wifi_qca/port/boards/lpc54018iotmodule/freertos/gt202"
+    "${lpc54018_dir}/middleware/wifi_qca/port/drivers/flexcomm_freertos"
+    "${lpc54018_dir}/middleware/wifi_qca/port/env/freertos"
+    "${lpc54018_dir}/middleware/usb/device"
+    "${lpc54018_dir}/middleware/usb/include"
+    "${lpc54018_dir}/middleware/usb/osa"
+    "${lpc54018_dir}/utilities"
+    "${lpc54018_dir}/utilities/io"
+    "${lpc54018_dir}/utilities/log"
+    "${lpc54018_dir}/utilities/str"
+    "${lpc54018_dir}/CMSIS/Include"
+)
+
+# Kernel
+afr_mcu_port(kernel)
+
+target_sources(
+    AFR::kernel::mcu_port
+    INTERFACE
+        ${lpc54018_src}
+        ${drivers_src}
+        ${mcuxpresso_src}
+        ${usb_device_src}
+        ${usb_osa_src}
+        ${wifi_qca_src}
+        ${utilities_src}
+        "${AFR_KERNEL_DIR}/portable/GCC/ARM_CM4F/port.c"
+        "${AFR_KERNEL_DIR}/portable/GCC/ARM_CM4F/portmacro.h"
+        "${AFR_KERNEL_DIR}/portable/MemMang/heap_4.c"
+)
+
+set(kernel_inc_dirs
+    ${lpc54018_includes}
+    "${AFR_KERNEL_DIR}/portable/GCC/ARM_CM4F"
+    "${AFR_KERNEL_DIR}/portable/CCS/ARM_CM3"
+    "${lpc54018_app_dir}/config_files"
+    "${lpc54018_app_dir}/application_code/nxp_code"
+    "$<IF:${AFR_IS_TESTING},${AFR_TESTS_DIR},${AFR_DEMOS_DIR}>/include"
```

Comment:
don't we need `${lpc54018_ports_dir}/<module>/include`? Like where is aws_wifi.h included?

Reply:
It is the common path: /amazon-freertos/demos/include

Not board specific as lpc54018_ports_dir

Reply:
it is in `AFR_Root/modules/ports/wifi/include` but I think it is included in the modules CMakeListst.txt